models/ed/R/veg2model.ED2.R

```diff
@@ -0,0 +1,234 @@
+      
+      # a series of checks to get pss info from settings if it exists
+      ### assuming one patch for now, otherwise these lines might change ###
+      time    <- ifelse(!is.null(inputinfo$time), inputinfo$time, start_year-1)
+      n.patch   <- ifelse(!is.null(inputinfo$patch), inputinfo$patch, 1)
+      trk     <- ifelse(!is.null(inputinfo$trk), inputinfo$trk, 1)
+      age     <- ifelse(!is.null(inputinfo$age), inputinfo$age, 100)
+      
+      obs <- data.frame(time = time, patch = n.patch, trk = trk, age = age)
+      
+      PEcAn.utils::logger.info(paste0("Values used in the patch file - time:", 
+                                      obs$time, ", patch:", obs$patch, ", trk:", 
+                                      obs$trk, ", age:", obs$age))
+     
+    }
+    
+    # this check might be redundant, direct nrows(obs) assignment could work 
+    # but leaving it for now
+    n.patch   <- ifelse(is.null(n.patch), nrow(obs), n.patch)
+      
+    ## fill missing data w/ defaults
+    obs$site  <- 1
+    obs$area  <- 1 / n.patch
+    obs$water <- 0
+    
+    # Reorder columns
+    obs <- obs[, c("site", "time", "patch", "trk", "age", "area", "water")]
+    
+    # Add soil data
+    soil            <- c(1, 5, 5, 0.01, 0, 1, 1)  #soil C & N pools (biogeochem) defaults (fsc,stsc,stsl,ssc,psc,msn,fsn)\t
+    soil.dat        <- as.data.frame(matrix(soil, n.patch, 7, byrow = TRUE))
+    names(soil.dat) <- c("fsc", "stsc", "stsl", "ssc", "psc", "msn", "fsn")
+    obs             <- cbind(obs, soil.dat)
+    
+    # Locally write files
+    write.table(obs, localfile, quote = FALSE, col.names = TRUE, row.names = FALSE)
+
+    
+  }else if(inputinfo$output == "css"){
+    
+    if(!is.null(inputinfo$path)){
+      obs <- read.table(inputinfo$path, header = TRUE)
+    }else if(is.null(obs)){
+      logger.severe("No css data found.")
+    }
+    
+    if(inputinfo$source == "FIA"){ # we still need one more check from pss
+      
+      pss <- read.table(runinfo$inputs$pss$path, header = TRUE)
+      
+      # Remove rows that don't map to any patch
+      obs <- obs[which(obs$patch %in% pss$patch), ]
+      if (nrow(obs) == 0) {
+        logger.severe("No trees map to previously selected patches.")
+      } else {
+        logger.debug(paste0(nrow(obs), " trees that map to previously selected patches."))
+      }
+    }
+    
+    if(is.null(obs$n)){
+      # is this how n is calculated?
+      obs$n <- ifelse(!is.null(inputinfo$area), nrow(obs)/inputinfo$area, 0.001)
+    }
+    
+    if(inputinfo$source == "GapMacro"){
+      inv.years <- as.numeric(unique(obs$year))
+      # suitable years
+      av.years <- inv.years[inv.years < start_year]
+      if(length(av.years) == 0){
+        logger.severe("No available years found in the data.")
+      }
+      obs$time <- max(av.years)
+      # filter out other years
+      obs <- obs[obs$year == obs$time, ]
+      colnames(obs)[colnames(obs) == "DBH"] <- "dbh"
```

Comment:
load_data converts to `DBH` which is the pecan name. 

On ED2 wiki, this column is named `dbh` (and it's what `fia2ED` was writing as well), I assumed it needs to be lower case `dbh` for ED to read properly, but now I remember something like these column headers didn't matter, just the order was important, is that correct?